docs/benchmarks.md

```diff
@@ -237,9 +237,35 @@ NVBENCH_BENCH_TYPES(benchmark, NVBENCH_TYPE_AXES(input_types, output_types))
 ```
 
 This would generate a total of 36 configurations and instantiate the benchmark 6
-times. Keep the rapid growth of these combinations in mind when choosing the
-number of values in an axis. See the section about combinatorial explosion for
-more examples and information.
+times.
+
+Keep the rapid growth of combinations due to multiple parameter axes in mind when
+choosing the number of values in an axis. See the section about combinatorial
+explosion for more examples and information.
+
+## Zipped/Tied Iteration of Value Axes
+
+At times multiple value axes need to be iterated like they are actually a tuple
+or zipped together. To enable this behavior you can request axes to be 'tied'
+together.
+
+```cpp
+// InputTypes: {char, int, unsigned int}
+// OutputTypes: {float, double}
+// NumInputs: {2^10, 2^20, 2^30}
+// Quality: {0.5, 1.0}
+
+using input_types = nvbench::type_list<char, int, unsigned int>;
+using output_types = nvbench::type_list<float, double>;
+NVBENCH_BENCH_TYPES(benchmark, NVBENCH_TYPE_AXES(input_types, output_types))
+  .set_type_axes_names({"InputType", "OutputType"})
+  .add_int64_axis("NumInputs", {1000, 10000, 100000, 200000, 200000, 200000})
+  .add_float64_axis("Quality", {0.05, 0.1, 0.25, 0.5, 0.75, 1.})
+  .zip_axes({"NumInputs", "Quality"});
```

Comment:
I would advocate for `add_zip_axis` to being the _only_ public way to add a zipped axis. Otherwise I think it will lead to confusion.

Reply:
I was initially concered that this would limit the ability of the `option_parser` to allow for runtime addition of `zips`.
But I believe we can do that via friend classes, and moving the `zip_axes` API to being protected on `benchmark_base`.

Overall I agree limiting the user facing API is best

Reply:
`add_zip_axes` constructs the axes and places them into the zip as an atomic operation, while `zip_axes` constructs a zip from existing axes.

Reply:
`add_zip_axis` is now the only public API